### ISSUE TITLE
React Content Query Webpart link in new window

### ISSUE THREAD

**Issue by Oermelie** [SharePoint/sp-dev-fx-webparts#866]
> Use the following form to submit an issue only if it's related to samples in this repo. If you have an issue related to the SharePoint Framework or its documentation, please submit the issue at [https://github.com/SharePoint/sp-dev-docs/issues/new](https://github.com/SharePoint/sp-dev-docs/issues/new). This will help us respond to your issue faster.
> 
> I Use the content Query webpart to search for webistes in a webiste library. I build a link
works perfect thanks.
but when i try to open the link in a new window with target="_blank" it doesnt work if the displayname of the link contains a space " ".

Thx
> 
> Thank you for reporting an issue or suggesting an enhancement. We appreciate your feedback - to help the team to understand your needs, please complete the below template to ensure we have the necessary details to assist you.
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

## Category

- [ ] Question
- [x] Bug
- [ ] Enhancement

> For the above list, an empty checkbox is [ ] as in <kbd>[</kbd><kbd>SPACE</kbd><kbd>]</kbd>. A checked checkbox is [x] with no space between the brackets. Use the `PREVIEW` tab at the top right to preview the rendering before submitting your issue.
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

## Authors

> Because of the way this repository is setup, samples authors do not get a notification when you create an issue. *It makes it less likely for you to get your issue resolved or to get help*. For the section above **@mention any author of the sample**. Authors' github handle can be found on the main sample documentation page, under the "solution" section. Use the `PREVIEW` tab at the top right to preview the rendering before submitting your issue.
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

## Expected or Desired Behavior

> _If you are reporting a bug, please describe the expected behavior. If you are suggesting an enhancement please describe thoroughly the enhancement, how it can be achieved, and expected benefit._
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

## Observed Behavior

> _If you are reporting a bug, please describe the behavior you expected to occur when performing the action. If you are making a suggestion, you can delete this section._
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

## Steps to Reproduce

> _If you are reporting a bug please describe the steps to reproduce the bug in sufficient detail to allow testing. Only way to fix things properly, is to have sufficient details to reproduce it. If you are making a suggestion, you can delete this section._
> 

I create links with the Content Query Webpart and add target="_blank" after the link in the <a> tag but when the following display name Contains a space " " it doesnt work!

**Comment by Oermelie** [comment]
Thx. I solved the problem by myself.
You have to put the displayname inside a <div> </div> Tag.
Thats it.
<a href="link" target="_blank"><div>This is my Link</div></a>

**Comment by Oermelie** [comment]
I found out that this only works if the target is a classic DisplayForm.
Try to link to a website page. Doesnt work

**Comment by spplante** [comment]
@Oermelie really not sure I understand the bug correctly there.

When you add the following to your template, it does not open google in a new window?!

<a href="https://google.com" target="_blank">Click me</a>

**Comment by matmig** [comment]
It works fine with <a href="https://google.com" target="_blank">Click me</a>
It doesn't work when you build dynamic href e.g.
<a href="https://google.com{{ID.textValue}}" target="_blank">Click me</a>

**Comment by spplante** [comment]
@matmig it does work when building dynamic href as well, however the target might be ignored when referring SharePoint modern pages and this has nothing to do with the CQWP, it is related to SharePoint intercepting clicks on links referring to modern pages in order to avoid reloading the whole page, it does the same thing when generating anchors from SPFx WebParts, see this article for the full explanation :

https://julieturner.net/2018/08/spfx-anchor-tags-hitting-the-target/

Basically just write your anchor the following way in your template and it will work :

```html
<a href="https://google.com{{ID.textValue}}" target="_blank" data-interception="off">Click me</a>
```

**Comment by matmig** [comment]
Thank you @spplante! I see it might have side effects but good to understand why it's happening.
